packages/lit-dev-content/site/tutorials/content/working-with-lists/03.md

```diff
@@ -0,0 +1,70 @@
+The rendering methods shown so far make it so that Lit keeps track of the
+DOM elements and will update them accordingly when the property items change.
+This is usually fine and efficient but sometimes it's preferable to control how
+each item is tied to the DOM element so that the DOM element is moved rather
+than its content updated.
+
+For instance, take a look at the example component presented here where each
+todo item is rendered as a list item with a checkbox. There are buttons to sort
+these items in ascending or descending alphabetical order. Check one of the
+items and change the sort order. Notice that the item's label text changes but
+not the position of the checked box. This is because Lit keeps track of the
+list of DOM nodes as they are and just updates the values that goes into the
+expression so only the label text changes.
+
+This is where the `repeat` directive can be of help. Using this directive with
+a key function lets Lit maintain the key-to-DOM association between updates
+by moving the generated DOM when required.
+
+To update the current implementaion, start by importing the `repeat` directive.
+
+{% switchable-sample %}
+
+```ts
+// my-element.ts
+import {repeat} from 'lit/directives/repeat.js';
+```
+
+```js
+// my-element.js
+import {repeat} from 'lit/directives/repeat.js';
+```
+
+{% endswitchable-sample %}
+
+Then use it in place of the `map` prototype method within the render, providing
+the iterable, a key function that returns a unique identifier for a particular
+item, and the template to render for each item.
+
+{% switchable-sample %}
+
+```ts
+// my-element.ts
+${repeat(
+  this.items,
+  (item) => item.id,
+  (item) => html`<li><label><input type="checkbox">${item.task}</label></li>`
+)}
+```
+
+```js
+// my-element.js
+${repeat(
+  this.items,
+  (item) => item.id,
+  (item) => html`<li><label><input type="checkbox">${item.task}</label></li>`
+)}
+```
+
+{% endswitchable-sample %}
+
+Confirm that this is working correctly by checking an item and changing the
+sort order. The check mark should now be moving to be next to the originally
+checked list item.
+
+<aside class="positive">
+The key function provided must return a unique key for an item. A key function
+that simply returns the index would have the same behavior as using
+<code>map</code>. See the <a href="/docs/templates/directives/#repeat">
+documentation</a> for more details.
+</aside>
```

Comment:
This fourth step is extremely good. I love that the initial code state is broken in a very clear way (which is described). Then the explanation and solution makes perfect sense and solves the original issue.

And TIL!